src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -264,16 +267,38 @@ private void putInternal(@Nullable String modelId, Model model, ActionListener<I
             indexRequestBuilder.setOpType(requestOpType); // Delegate whether this request can update based on opType
             indexRequestBuilder.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
 
-            // After metadata update finishes, remove item from cache if necessary. If no model id is
+            // After metadata update finishes, remove item from every node's cache if necessary. If no model id is
             // passed then nothing needs to be removed from the cache
-            //TODO: Bug. Model needs to be removed from all nodes caches, not just local.
-            // https://github.com/opensearch-project/k-NN/issues/93
             ActionListener<IndexResponse> onMetaListener;
             if (modelId != null) {
-                onMetaListener = ActionListener.wrap(response -> {
-                    ModelCache.getInstance().remove(modelId);
-                    listener.onResponse(response);
-                }, listener::onFailure);
+                onMetaListener = ActionListener.wrap(indexResponse -> client.execute(
+                        RemoveModelFromCacheAction.INSTANCE,
+                        new RemoveModelFromCacheRequest(modelId),
+                        ActionListener.wrap(
+                                removeModelFromCacheResponse -> {
+
+                                    if (removeModelFromCacheResponse.hasFailures()) {
```

Comment:
Shall we flip if condition and return if no failure to improve readability?

Reply:
Make sense.